internal/deployments.go

```diff
@@ -229,49 +237,129 @@ func storageRequest(job *model.Job) resourcev1.Quantity {
 	return value
 }
 
+// inputStagingContainer returns the init container to be used for staging input files. This init container
+// is only used when iRODS CSI driver integration is disabled.
+func (i *Internal) inputStagingContainer(job *model.Job) apiv1.Container {
+	return apiv1.Container{
+		Name:            fileTransfersInitContainerName,
+		Image:           fmt.Sprintf("%s:%s", i.PorklockImage, i.PorklockTag),
+		Command:         append(fileTransferCommand(job), "--no-service"),
+		ImagePullPolicy: apiv1.PullPolicy(apiv1.PullAlways),
+		WorkingDir:      inputPathListMountPath,
+		VolumeMounts:    i.fileTransfersVolumeMounts(job),
+		Ports: []apiv1.ContainerPort{
+			{
+				Name:          fileTransfersPortName,
+				ContainerPort: fileTransfersPort,
+				Protocol:      apiv1.Protocol("TCP"),
+			},
+		},
+		SecurityContext: &apiv1.SecurityContext{
+			RunAsUser:  int64Ptr(int64(job.Steps[0].Component.Container.UID)),
+			RunAsGroup: int64Ptr(int64(job.Steps[0].Component.Container.UID)),
+			Capabilities: &apiv1.Capabilities{
+				Drop: []apiv1.Capability{
+					"SETPCAP",
+					"AUDIT_WRITE",
+					"KILL",
+					"SETGID",
+					"SETUID",
+					"NET_BIND_SERVICE",
+					"SYS_CHROOT",
+					"SETFCAP",
+					"FSETID",
+					"NET_RAW",
+					"MKNOD",
+				},
+			},
+		},
+	}
+}
+
+// workingDirPrepContainer returns the init container to be used for preparing the working directory volume
+// for use within the VICE analysis. This init container is only used when iRODS CSI driver integration is
+// enabled.
+//
+// It may seem odd to use the file transfer image to initialize the working directory when no files are actually
+// being transferred, but it works. We use it for a couple of different reasons. First, we need a Unix shell and
+// it has one. Second, it's already set up so that we can configure it in a way that avoids image pull rate limits.
+func (i *Internal) workingDirPrepContainer(job *model.Job) (apiv1.Container, error) {
+
+	// Determine the iRODS zone name.
+	zone, err := i.getIRODSZone(job.UserHome)
```

Comment:
This may not matter, but I'm wondering if it'd be better to pull the irods zone out of the config file. The irods zone is in jobservices.yml already, but none of the irods config stuff is used by app-exposer right now.

I suppose the other question is, as far as I can see, this `zone` is only used below to stick it into `ln -s "/<zone>/home" .` -- could we just do `fmt.Sprintf("ln -s \"%s\" .", job.UserHome)` instead and drop this error case entirely?

Reply:
I do like the idea of eliminating the error case. I believe that the user home contains the username as well, and I'd like the symbolic link to point to the parent directory. I could do a regular expression replacement to avoid that, though. I'll throw that change in.

Reply:
On second thought, I think I'll probably grab the zone out of the configuration file. That seems to be the simplest option to me at the moment. That will allow me to avoid doing any error checking on the job submission in this function.